### ISSUE TITLE
Unable to sample a simple IRT model

### ISSUE THREAD

**Issue by joshchang** [tensorflow/probability#505]
I apologize in advance for being new to TF/TFP... I also posted a variant of this issue as a comment on #69 

 I am trying to learn TFP by implementing the graded response model for polytomous item response theory, which I previously have used Stan to sample.

 I have implemented the model here: https://colab.research.google.com/drive/1ZAmkuQF2XoV0Jy9_EJtGTJrD3Tn4UlAJ

The model is defined:

```python
"""
Probabilities for a single person
discriminations   I (item)
difficulties I x K-1
abilities N 

returns
prob N x I x K
"""
def grm_model_prob(abilities, discriminations, difficulties, expanded=False):
    offsets = difficulties[tf.newaxis,:,:] - abilities[:, tf.newaxis, tf.newaxis]
    scaled = offsets*discriminations[tf.newaxis, :, tf.newaxis]
    logits = 1.0/(1+tf.exp(scaled))
    logits = tf.pad(logits, paddings=(
        (0, 0), (0, 0), (1, 0)), mode='constant', constant_values=1.)
    logits = tf.pad(logits, paddings=(
        (0, 0), (0, 0), (0, 1)), mode='constant', constant_values=0.)
    probs = logits[:, :, :-1] - logits[:, :, 1:]
    
    return probs

"""
Probabilities for single items
discriminations   I (item)
difficulties  I x K - 1
abilities N

mu (difficulty local) I
tau (difficulty) I x K-2

"""
def joint_log_prob(responses, discriminations, difficulties0, ddifficulties, abilities, mu):
    # assemble the difficulties
    d0 = tf.concat([difficulties0[:,tf.newaxis],ddifficulties],axis=1)
    difficulties = tf.cumsum(d0,axis=1)
    return tf.reduce_sum(log_likelihood(responses, discriminations, difficulties, abilities)) + \
        joint_log_prior(discriminations, difficulties0, ddifficulties, abilities, mu)

def log_likelihood(responses, discriminations, difficulties, abilities):
    rv_responses = tfd.Categorical(grm_model_prob(
        abilities, discriminations, difficulties))
    return rv_responses.log_prob(responses)

def joint_log_prior(discriminations, difficulties0, ddifficulties, abilities, mu):
    rv_discriminations = tfd.HalfNormal(scale=tf.ones_like(discriminations))
    rv_difficulties0 =  tfd.Normal(loc=mu, scale=3.*tf.ones_like(difficulties0))
    rv_ddifficulties =  tfd.HalfNormal(scale=tf.ones_like(ddifficulties))
    rv_abilities =  tfd.Normal(loc=tf.zeros_like(abilities), scale=3.)
    rv_mu =  tfd.Normal(loc=tf.zeros_like(mu),scale=3.)
    
    return tf.reduce_sum(rv_discriminations.log_prob(discriminations)) + \
        tf.reduce_sum(rv_difficulties0.log_prob(difficulties0)) + \
        tf.reduce_sum(rv_ddifficulties.log_prob(ddifficulties)) + \
        tf.reduce_sum(rv_abilities.log_prob(abilities))

```

I *think* this code correctly represents the simple hierarchical model typeset in the colab. However, I can get neither MH or HMC to sample this model. They both evaluate to acceptance ratio of 0, and it seems that HMC isn't able to compute gradients for the model.

 Am I doing something obviously wrong? Thanks in advance.

**Comment by joshchang** [comment]
Looks like adding step size adaptation helps

```python

number_of_steps=4000 
burnin=3000 
num_leapfrog_steps=5
hmc=tfp.mcmc.TransformedTransitionKernel(
    inner_kernel=tfp.mcmc.HamiltonianMonteCarlo(
        target_log_prob_fn=unnormalized_posterior_log_prob,
        num_leapfrog_steps=num_leapfrog_steps,
        step_size=0.1,
        state_gradients_are_stopped=True),
    bijector=unconstraining_bijectors)

mh = tfp.mcmc.TransformedTransitionKernel(
    inner_kernel=tfp.mcmc.RandomWalkMetropolis(
        target_log_prob_fn=unnormalized_posterior_log_prob),
    bijector=unconstraining_bijectors)

kernel = tfp.mcmc.SimpleStepSizeAdaptation(
    inner_kernel=hmc, num_adaptation_steps=int(burnin * 0.8))

```
